File: app/src/main/java/cc/tomko/outify/OutifyApplication.java
package cc.tomko.outify;

import android.app.Application;

import cc.tomko.outify.core.Session;
import cc.tomko.outify.core.AuthManager;
import cc.tomko.outify.core.spirc.Spirc;
import cc.tomko.outify.playback.AudioManager;
import cc.tomko.outify.playback.AudioPlayer;

import com.google.crypto.tink.aead.AeadConfig;

public class OutifyApplication extends Application {
    public static SecureStorage secureStorage;

    public static AuthManager spAuthManager;
    public static AudioManager audioManager;
    public static AudioPlayer audioPlayer;
    public static Session session;
    public static Spirc spirc;

    @Override
    public void onCreate() {
        super.onCreate();

        System.loadLibrary("librespot_ffi");
        LibrespotFfi.libInit(getApplicationContext());

        try {
            AeadConfig.register();
            secureStorage = new SecureStorage(getApplicationContext());
        } catch (Exception e) {
            throw new RuntimeException(e);
        }

        audioPlayer = new AudioPlayer();

        session = new Session();
//        session.initializeSession();

//        try {
//            debug.debug1(secureStorage.getString(SecureStorage.Keys.ACCESS_TOKEN));
//        } catch (GeneralSecurityException e) {
//            throw new RuntimeException(e);
//        }

        audioManager = new AudioManager();

        // Initializing SpAuthManager
        spAuthManager = new AuthManager();
    }
}

File: app/src/main/java/cc/tomko/outify/OutifyApplication.kt
package cc.tomko.outify

import android.app.Application
import androidx.media3.common.util.UnstableApi
import cc.tomko.outify.core.spirc.SpircController
import cc.tomko.outify.data.database.AppDatabase
import cc.tomko.outify.ui.viewmodel.detail.DetailViewModelStore
import cc.tomko.outify.ui.viewmodel.detail.setDetailViewModelStore
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Inject

const val ALBUM_COVER_URL: String = "https://i.scdn.co/image/"

@HiltAndroidApp
class OutifyApplication : Application() {
    lateinit var database: AppDatabase
        private set

    @Inject
    lateinit var spircController: SpircController

    @Inject
    lateinit var detailViewModelStore: DetailViewModelStore

    @UnstableApi
    override fun onCreate() {
        super.onCreate()
        database = AppDatabase.getInstance(this)

        setDetailViewModelStore(detailViewModelStore)

        System.loadLibrary("librespot_ffi")
        LibrespotFfi.libInit(applicationContext)

        spircController.start()

//        AeadConfig.register()
    }
}